File: data structures and algorithms/cpp/06_120.cpp
std::vector<int> solution(const std::vector<int>& a) {
    int n = a.size();
    std::vector<int> result(n);
    
    for (int i = 0; i < n; i++) {
        int leftVal = (i-1 < 0) ? 0 : a[i-1];
        int rightVal = (i+1 >= n) ? 0 : a[i+1];
        result[i] = leftVal + a[i] + rightVal;
    }
    
    return result;
}

File: data structures and algorithms/cs/06_120.cs
public class ArrayMutation {
    public static int[] Solution(int n, int[] a) {
        int[] b = new int[n];
        
        for (int i = 0; i < n; i++) {
            // Left neighbor (or 0 if out of bounds)
            int left = (i > 0) ? a[i-1] : 0;
            
            // Current element
            int current = a[i];
            
            // Right neighbor (or 0 if out of bounds)
            int right = (i < n-1) ? a[i+1] : 0;
            
            // Sum all three values
            b[i] = left + current + right;
        }
        
        return b;
    }
    
    public static void Main(string[] args) {
        int[] a = {4, 0, 1, -2, 3};
        int n = a.Length;
        int[] result = Solution(n, a);
        
        // Print the result
        foreach (int val in result) {
            Console.Write(val + " ");
        }
        // Output: 4 5 -1 2 1
    }
}